uni/lib/view/common_widgets/random_image.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+
+class RotatingImage extends StatefulWidget {
+  final List<String> imagePaths;
+  final double width;
+  final double height;
+
+  const RotatingImage({required this.imagePaths, required this.width, required this.height, Key? key}) : super(key: key);
+
+  @override
+  State<RotatingImage> createState() => _RotatingImageState();
+}
+
+class _RotatingImageState extends State<RotatingImage> {
+  int _index = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    Timer.periodic(const Duration(minutes: 1), (timer) {
+      setState(() {
+        _index = (_index + 1) % widget.imagePaths.length;
+      });
+    });
```

Comment:
This will not effectively change the image, since this widget instance is destroyed when no longer on screen. I think that saving the `rotatingImageIndex` globally on the shared preferences, with last update time, and doing (+1 % len) when building this widget should be fine.